### PR TITLE
samples: common: dtm: radio_test: Extract common FEM functions

### PR DESCRIPTION
**Pull request by dchat-nordic**
Extarct FEM functionality common to dtm and radio_test to a separate directory.

### CODE REVIEW

**Comment by b-gent** [comment]
@wbober  can you confirm that adding a 'Common samples' section is fine (regarding the discussion about sample folder structure that we had). Thanks.

**Comment by peknis** [comment]
> @wbober can you confirm that adding a 'Common samples' section is fine (regarding the discussion about sample folder structure that we had). Thanks.

Actually, I believe this should be a library, not a sample, after all. But in a new section "Common libraries".

**Comment by wbober** [comment]
@carlescufi Could you please advise on if this should go to `subsys` or `lib`?

File: doc/nrf/libraries/fem_al/index.rst [hunk 1]
```diff
@@ -0,0 +1,48 @@
+.. _fem_al_lib:
+
+FEM Abstraction Layer
```
Comment (peknis) [inline doc/nrf/libraries/fem_al/index.rst:3]: Not so sure about the capitalization. Maybe this is fine with  lower case.

File: doc/nrf/libraries/fem_al/index.rst [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+#####################
+
+.. contents::
+   :local:
+   :depth: 2
+
+The library is a wrapper for the MPSL FEM.
```
Comment (peknis) [inline doc/nrf/libraries/fem_al/index.rst:10]:
```suggestion
This library is a wrapper for the MPSL FEM.
```

File: doc/nrf/libraries/fem_al/index.rst [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+Overview
+********
+
+The library is designed to simplify low level usage of a FEM hardware.
```
Comment (peknis) [inline doc/nrf/libraries/fem_al/index.rst:15]:
```suggestion
The library simplifies low level usage of FEM hardware.
```

File: doc/nrf/libraries/fem_al/index.rst [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+Requirements
+************
+
+The library requires the usage of a FEM and MPSL functionality.
```
Comment (peknis) [inline doc/nrf/libraries/fem_al/index.rst:20]: This does not sound like a proper requirement, as the lib is meant to be used with FEM and MPSL, which is stated already before. If there are no other requirements, you might as well leave this section out.

File: doc/nrf/libraries/fem_al/index.rst [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+
+Configuration
+*************
+
+The library is enabled by the :kconfig:option:`CONFIG_FEM_AL_LIB`.
```
Comment (peknis) [inline doc/nrf/libraries/fem_al/index.rst:25]:
```suggestion
To enable this library, use the Kconfig option ``CONFIG_FEM_AL_LIB``.
```
As this option is not available in the Kconfig reference, you should not use the :kconfig:option: tag.

File: doc/nrf/libraries/fem_al/index.rst [hunk 6]
```diff
@@ -0,0 +1,48 @@
+.. _fem_al_lib:
```
Comment (peknis) [inline doc/nrf/libraries/fem_al/index.rst:1]: If you want to add this lib doc to the Other libraries section, place it in doc/nrf/libraries/other and rename this file to fem_al.rst, for example

**Review by peknis — CHANGES_REQUESTED** [review]

File: samples/peripheral/radio_test/README.rst [hunk 7]
```diff
@@ -243,6 +243,7 @@ Dependencies
 This sample uses the following |NCS| library:
```
Comment (peknis) [inline samples/peripheral/radio_test/README.rst:243]:
```suggestion
This sample uses the following |NCS| libraries:
```

**Review by peknis — CHANGES_REQUESTED** [review]

**Review by peknis — APPROVED** [review]

**Review by maje-emb — APPROVED** [review]

**Comment by carlescufi** [comment]
> @carlescufi Could you please advise on if this should go to `subsys` or `lib`?

Personally in this case I think this belongs in `lib`, @dchat-nordic. The distinction is not always clear, but this seems "passive" (i.e. no startup or threading).

**Review by carlescufi — CHANGES_REQUESTED** [review]
Please move to `lib`

**Comment by wbober** [comment]
LGTM

**Review by carlescufi — APPROVED** [review]

**Pull request merged**